grype/version/semantic_constraint.go

```diff
@@ -35,7 +35,7 @@ func newSemanticConstraint(constStr string) (semanticConstraint, error) {
 }
 
 func (c semanticConstraint) supported(format Format) bool {
-	return format == SemanticFormat
+	return format == SemanticFormat || format == GemfileFormat
```

Comment:
since this feels a bit odd, it would be worth adding a comment here as to why there is an ecosystem exception made a the generic semver method.